pkg/services/resource.go

```diff
@@ -56,6 +56,11 @@ func (s *sqlResourceService) Get(ctx context.Context, id string) (*api.Resource,
 }
 
 func (s *sqlResourceService) Create(ctx context.Context, resource *api.Resource) (*api.Resource, *errors.ServiceError) {
+	if resource.Name != "" {
+		if err := ValidateResourceName(resource); err != nil {
+			return nil, errors.Validation("the name in the resource is invalid, %v", err)
+		}
+	}
```

Comment:
should we also check the name immutable when update?

Reply:
Name string `gorm:"uniqueIndex"` there is error from database. we do not need to check here again for updating.

Reply:
the name is not exposed for patch. so the name will be ignored if it is changed when patch.